tests/test_utils.py

```diff
@@ -0,0 +1,136 @@
+    make_test_exposure,
+    make_test_injection_catalog,
+    make_test_reference_pipeline,
+)
+from lsst.utils.tests import MemoryTestCase, TestCase
+
+TEST_DIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class SourceInjectionUtilsTestCase(TestCase):
+    """Test the utility functions in the source_injection package."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.root = makeTestTempDir(TEST_DIR)
+        cls.creator_butler = makeTestRepo(cls.root)
+        cls.writeable_butler = makeTestCollection(cls.creator_butler)
+
+    @classmethod
+    def tearDownClass(cls):
+        del cls.writeable_butler
+        del cls.creator_butler
+        removeTestTempDir(cls.root)
+
+    def setUp(self):
+        self.exposure = make_test_exposure()
+        self.injection_catalog = make_test_injection_catalog(
+            self.exposure.getWcs(),
+            self.exposure.getBBox(),
+        )
+        self.reference_pipeline = make_test_reference_pipeline()
+
+    def tearDown(self):
+        del self.exposure
+        del self.injection_catalog
+        del self.reference_pipeline
+
+    def test_generate_injection_catalog(self):
+        self.assertEqual(len(self.injection_catalog), 30)
+        expected_columns = {"injection_id", "ra", "dec", "source_type", "mag"}
+        self.assertEqual(set(self.injection_catalog.columns), expected_columns)
+
+    def test_make_injection_pipeline(self):
+        injection_pipeline = Pipeline("reference_pipeline")
+        injection_pipeline.addTask(ExposureInjectTask, "inject_exposure")
+        merged_pipeline = make_injection_pipeline(
+            dataset_type_name="postISRCCD",
+            reference_pipeline=self.reference_pipeline,
+            injection_pipeline=injection_pipeline,
+            exclude_subsets=False,
+            prefix="injected_",
+            instrument="lsst.obs.subaru.HyperSuprimeCam",
+            log_level=logging.DEBUG,
+        )
+        expanded_pipeline = merged_pipeline.toExpandedPipeline()
+        expected_subset_tasks = ["isr", "inject_exposure", "characterizeImage"]
+        merged_task_subsets = [merged_pipeline.findSubsetsWithLabel(x) for x in expected_subset_tasks]
+        self.assertEqual(len(merged_task_subsets), len(expected_subset_tasks))
+        for taskDef in expanded_pipeline:
+            conns = taskDef.connections
+            if taskDef.label == "isr":
+                self.assertEqual(conns.outputExposure.name, "postISRCCD")
+            elif taskDef.label == "inject_exposure":
+                self.assertEqual(conns.input_exposure.name, "postISRCCD")
+                self.assertEqual(conns.output_exposure.name, "injected_postISRCCD")
+                self.assertEqual(conns.output_catalog.name, "injected_postISRCCD_catalog")
+            elif taskDef.label == "characterizeImage":
+                self.assertEqual(conns.exposure.name, "injected_postISRCCD")
+                self.assertEqual(conns.characterized.name, "injected_icExp")
```

Comment:
Check for `injected_icSrc` too?

Reply:
Added `injected_icSrc` and `injected_icExpBackground` as well.